### PR TITLE
[WIP] [v6r21?] Replacing pyGSI with M2Crypto

### PR DESCRIPTION
**Pull request by miloszz**
#3469 got extremely messy...

BEGINRELEASENOTES
Thank you for writing the text to appear in the release notes. It will show up
exactly as it appears between the two bold lines

Please follow the template:
*Subsystem
NEW/CHANGE/FIX: explanation

For examples look into release.notes

ENDRELEASENOTES

### CODE REVIEW

**Comment by chaen** [comment]
If you look at the diff, many many lines are duplicated

File: Core/Security/X509Chain.py [hunk 1]
```diff
@@ -58,8 +57,9 @@ def loadChainFromFile(self, chainLocation):
     Return : S_OK / S_ERROR
     """
     try:
-      with file(chainLocation) as fd:
-        pemData = fd.read()
+      fd = file(chainLocation)
+      pemData = fd.read()
+      fd.close()
```
Comment (fstagni) [inline Core/Security/X509Chain.py]: This was probably changed by some other PRs.

File: Core/Security/X509Chain.py [hunk 2]
```diff
@@ -96,8 +103,9 @@ def loadKeyFromFile(self, chainLocation, password=False):
...
```
Comment (fstagni) [inline Core/Security/X509Chain.py]: This too.

File: Core/Security/X509Chain.py [hunk 3]
```diff
@@ -412,15 +404,11 @@ def __checkProxyDN(self, certStep, issuerStep):
         if ext.get_sn() == "proxyCertInfo":
           contraint = [line.split(":")[1].strip() for line in ext.get_value().split("\n")
                        if line.split(":")[0] == "Path Length Constraint"]
-          if not contraint:
+          if len(contraint) == 0:
```
Comment (fstagni) [inline Core/Security/X509Chain.py]: This too.

File: Core/Security/X509Chain.py [hunk 4]
```diff
@@ -446,13 +434,14 @@ def getDIRACGroup(self, ignoreDefault=False):
...
-    # No DIRAC group found, try to get the default one
-    return self.getCertInChain(self.__firstProxyStep)['Value'].getDIRACGroup(ignoreDefault=ignoreDefault)
+    else:
+      # The code below will find the first match of the DIRAC group
+      for i in range(len(self.__certList) - 1, -1, -1):
+        retVal = self.getCertInChain(i)['Value'].getDIRACGroup(ignoreDefault=True)
+        if retVal['OK'] and 'Value' in retVal and retVal['Value']:
+          return retVal
+      # No DIRAC group found, try to get the default one
+      return self.getCertInChain(self.__firstProxyStep)['Value'].getDIRACGroup(ignoreDefault=ignoreDefault)
```
Comment (fstagni) [inline Core/Security/X509Chain.py]: This too.

File: FrameworkSystem/Client/ProxyGeneration.py [hunk 5]
```diff
@@ -28,243 +29,241 @@ class CLIParams(object):
...
 
-  def setStdinPasswd( self, _arg ):
+  def setStdinPasswd(self, _arg):
     self.stdinPasswd = True
     return S_OK()
 
-  def setStrict( self, _arg ):
+  def setStrict(self, _arg):
     self.strict = True
     return S_OK()
```
Comment (fstagni) [inline FrameworkSystem/Client/ProxyGeneration.py]: ...it's time to remove all these getters and setters methods!

**Comment by fstagni** [comment]
Thanks for restarting this. Please check the travis errors.

**Comment by chaen** [comment]
hi @miloszz , how's the rebasing going ?

**Comment by miloszz** [comment]
My ssd literally died today. I'm setting up fresh system install right now...

**Comment by chaen** [comment]
Hi @miloszz 
tough luck ! I hope you did not lose too much. Did you make any progress ?

**Comment by miloszz** [comment]
Mess again. I'll get back to this on Thursday.

**Comment by fstagni** [comment]
This PR is still screwed up. Apart from having 250+ commits, it contains changes for 179 files, and a fraction (presumably large) of the proposed changes have nothing to do with pyGSI and M2Crypto.

**Comment by fstagni** [comment]
Closing, @chaen taking care (comments still welcome)

**Pull request closed without merging**